lib/rubocop/cop/rspec/aggregate_examples.rb

```diff
@@ -0,0 +1,294 @@
+          :validate_length_of
+          :validate_inclusion_of
+          :validates_exclusion_of
+        ].freeze
+
+        def on_block(node)
+          example_group_with_several_examples(node) do |all_examples|
+            examples = aggregateable_examples(all_examples)
+            next if examples.count < 2
+
+            message = message_for(examples)
+            add_offense(examples[1], location: :expression, message: message)
+          end
+        end
+
+        def autocorrect(example_node)
+          node = example_node.parent
+          examples = aggregateable_examples(node.each_child_node(:block))
+            .select { |child| examples_for_autocorrect?(child) }
+          return if examples.count < 2
+
+          replacement = aggregated_example(examples)
+          lambda do |corrector|
+            range = range_for_replace(examples)
+            corrector.replace(range, replacement)
+            examples[1..-1].each { |example| drop_example(corrector, example) }
+          end
+        end
+
+        private
+
+        def_node_matcher :example_group_with_several_examples, <<-PATTERN
+          (block
+            (send { nil? (const nil? :RSpec) } { :describe :context :feature :example_group } ...)
+            _
+            (begin $...)
+          )
+        PATTERN
+
+        def aggregateable_examples(examples)
+          examples
+            .reject { |example| has_metadata?(example) }
+            .select { |example| example_with_expectations_solely?(example) }
+        end
+
+        def range_for_replace(examples)
+          range = range_by_whole_lines(examples.first.source_range,
+                                       include_final_newline: true)
+          next_range = range_by_whole_lines(examples[1].source_range)
+          range = range.resize(range.length + 1) if adjacent?(range, next_range)
+          range
+        end
+
+        def adjacent?(range, another_range)
+          range.end_pos + 1 == another_range.begin_pos
+        end
+
+        def aggregated_example(examples)
+          base_indent = ' ' * examples.first.source_range.column
+          [
+            "#{base_indent}specify do",
+            *examples.map { |example| transform_body(example, base_indent) },
+            "#{base_indent}end\n"
+          ].join("\n")
+        end
+
+        def drop_example(corrector, example)
+          aggregated_range = range_by_whole_lines(example.source_range,
+                                                  include_final_newline: true)
+          corrector.remove(aggregated_range)
+        end
+
+        # Extracts and transforms the body.
+        # `its(:something) { is_expected.to ... }` is a special case, since
+        # it's impossible to aggregate its body as is,
+        # it needs to be converted to `expect(subject.something).to ...`
+        # Additionally indents the example code properly.
+        def transform_body(node, base_indent)
+          method, _args, body = *node
```

Comment:
you can use `node.method_name` and `node.body` and avoid manual deconstruction - block nodes delegate method_name to the send node